client/src/test/java/io/moquette/spi/impl/security/AESTest.java

```diff
@@ -0,0 +1,143 @@
+        Cipher cipher = PowerMockito.mock(Cipher.class);
+        byte[] myByteArray = {(byte) 0, (byte) 0, (byte) 0, (byte) -32, (byte) 0};
+
+        Method doFinalMethod =
+            DTUMemberMatcher.method(Cipher.class, "doFinal", byte[].class);
+        PowerMockito.doReturn(myByteArray)
+            .when(cipher, doFinalMethod)
+            .withArguments(or(isA(byte[].class), isNull(byte[].class)));
+
+        Method getInstanceMethod =
+            DTUMemberMatcher.method(Cipher.class, "getInstance", String.class);
+        PowerMockito.doReturn(cipher)
+            .when(Cipher.class, getInstanceMethod)
+            .withArguments(or(isA(String.class), isNull(String.class)));
+        PowerMockito.when(System.currentTimeMillis()).thenReturn(1_515_585_600_000L);
+
+        Assert.assertNull(AES.AESDecrypt(null, (byte[]) null, true));
+    }
+
+    @PrepareForTest({Cipher.class, AES.class, System.class})
+    @Test
+    public void testAESDecrypt3() throws Exception {
+        PowerMockito.mockStatic(System.class);
+        PowerMockito.mockStatic(Cipher.class);
+        Cipher cipher = PowerMockito.mock(Cipher.class);
+        byte[] myByteArray = {(byte) 0, (byte) 0, (byte) 0, (byte) 1, (byte) 0};
+
+        Method doFinalMethod =
+            DTUMemberMatcher.method(Cipher.class, "doFinal", byte[].class);
+        PowerMockito.doReturn(myByteArray)
+            .when(cipher, doFinalMethod)
+            .withArguments(or(isA(byte[].class), isNull(byte[].class)));
+
+        Method getInstanceMethod =
+            DTUMemberMatcher.method(Cipher.class, "getInstance", String.class);
+        PowerMockito.doReturn(cipher)
+            .when(Cipher.class, getInstanceMethod)
+            .withArguments(or(isA(String.class), isNull(String.class)));
+        PowerMockito.when(System.currentTimeMillis()).thenReturn(1_515_585_600_000L);
+
+        Assert.assertArrayEquals(new byte[]{(byte) 0},
+            AES.AESDecrypt(null, (byte[]) null, false));
+    }
+
+    @PrepareForTest({Cipher.class, AES.class})
+    @Test
+    public void testAESDecrypt4() throws Exception {
+        PowerMockito.mockStatic(Cipher.class);
+        Cipher cipher = PowerMockito.mock(Cipher.class);
+        byte[] myByteArray = {(byte) 0, (byte) 0, (byte) 0, (byte) 0};
+
+        Method doFinalMethod =
+            DTUMemberMatcher.method(Cipher.class, "doFinal", byte[].class);
+        PowerMockito.doReturn(myByteArray)
+            .when(cipher, doFinalMethod)
+            .withArguments(or(isA(byte[].class), isNull(byte[].class)));
+
+        Method getInstanceMethod =
+            DTUMemberMatcher.method(Cipher.class, "getInstance", String.class);
+        PowerMockito.doReturn(cipher)
+            .when(Cipher.class, getInstanceMethod)
+            .withArguments(or(isA(String.class), isNull(String.class)));
+
+        Assert.assertNull(AES.AESDecrypt(null, (byte[]) null, false));
+    }
+
+    @PrepareForTest({Cipher.class, AES.class, System.class})
```

Comment:
Is the `System.class` required for this test?

Reply:
Removed